include/vast/Interfaces/AST/DeclInterface.td

```diff
@@ -0,0 +1,92 @@
+    let description = [{An interface to provide [`clang::Decl`]
+(https://clang.llvm.org/doxygen/classclang_1_1Decl.html) functionality.}];
+
+    let methods = [
+        InterfaceMethod< "clang::Decl::isImplicit", "bool", "isImplicit", (ins) >,
+        InterfaceMethod< "clang::Decl::getDeclContext", "void", "getDeclContext", (ins) >,
+        InterfaceMethod< "clang::Decl::getASTContext", "void", "getASTContext", (ins) >
+    ];
+}
+
+def VAST_NamedDeclInterface : VAST_ASTOpInterface< "VAST_NamedDeclInterface",
+    [VAST_DeclInterface] > {
+
+    let description = [{An interface to provide [`clang::NamedDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1NamedDecl.html) functionality.}];
+
+    let methods = [];
+}
+
+def VAST_ValueDeclInterface : VAST_ASTOpInterface< "VAST_ValueDeclInterface",
+    [VAST_NamedDeclInterface] > {
+
+    let description = [{An interface to provide [`clang::ValueDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1ValueDecl.html) functionality.}];
+
+    let methods = [
+        InterfaceMethod< "clang::ValueDecl::getType", "void", "getType", (ins) >
+    ];
+}
+
+def VAST_DeclaratorDeclInterface : VAST_ASTOpInterface< "VAST_DeclaratorDeclInterface",
+    [VAST_ValueDeclInterface] > {
+    
+    let description = [{An interface to provide [`clang::DeclaratorDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1DeclaratorDecl.html) functionality.}];
+
+    let methods = [];
+}
+
+def VAST_FieldDeclInterface : VAST_ASTOpInterface< "VAST_FieldDeclInterface",
+    [VAST_DeclaratorDeclInterface] > {
+    let description = [{An interface to provide [`clang::FieldDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1DeclaratorDecl.html) functionality.}];
+
+    let methods = [
+        InterfaceMethod< "clang::FieldDecl::isUnnamedBitField", "bool", "isUnnamedBitField", (ins) >,
+        InterfaceMethod< "clang::FieldDecl::isZeroSize", "bool", "isZeroSize", (ins) >
+    ];
+}
+
+def VAST_VarDeclInterface : VAST_ASTOpInterface< "VAST_VarDeclInterface",
+    [VAST_DeclaratorDeclInterface] > {
+    let description = [{An interface to provide [`clang::VarDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1VarDecl.html) functionality.}];
+
+    let methods = [
+        InterfaceMethod< "clang::VarDecl::isLocalVarDecl", "bool", "isLocalVarDecl", (ins) >,
+        InterfaceMethod< "clang::VarDecl::hasGlobalStorage", "bool", "hasGlobalStorage", (ins) >,
+        InterfaceMethod< "clang::VarDecl::isExceptionVariable", "bool", "isExceptionVariable", (ins) >,
+        InterfaceMethod< "clang::VarDecl::isInitCapture", "bool", "isInitCapture", (ins) >,
+        InterfaceMethod< "clang::VarDecl::getInit", "void", "getInit", (ins) >
+    ];
+}
+
+def VAST_RecordDeclInterface : VAST_ASTOpInterface< "VAST_RecordDeclInterface",
+    [VAST_TagDeclInterface] > {
+    let description = [{An interface to provide [`clang::RecordDecl`]
+(https://clang.llvm.org/doxygen/classclang_1_1RecordDecl.html) functionality.}];
```

Comment:
Indent this and the similars?

Reply:
All indentation problems have been fixed.